src/redteam/index.ts

```diff
@@ -379,6 +381,22 @@ export async function synthesize({
     }
   }
 
+  // Check API health before proceeding
+  if (shouldGenerateRemote()) {
+    const healthUrl = getRemoteHealthUrl();
+    if (healthUrl) {
+      logger.debug('Checking promptfoo API health...');
+      const healthResult = await checkRemoteHealth(healthUrl);
+      if (healthResult.status !== 'OK') {
+        throw new Error(
+          `Unable to proceed with test generation: ${healthResult.message}\n` +
+            'Please check your API configuration or try again later.',
+        );
+      }
+      logger.debug('API health check passed');
+    }
+  }
```

Comment:
If they're going to hit the unaligned provider we should check that they have access to that as well. There is a function to get that URL in https://github.com/promptfoo/promptfoo/blob/main/src/redteam/remoteGeneration.ts as well.